stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpService.java

```diff
@@ -114,4 +157,86 @@ public void setCorsHeaders(CorsHeaders corsHeaders) {
         }
         return uriTemplate;
     }
+
+    public Struct getWebSocketUpgradeConfig() {
+        return webSocketUpgradeConfig;
+    }
+
+    public void setWebSocketUpgradeConfig(Struct webSocketUpgradeConfig) {
+        this.webSocketUpgradeConfig = webSocketUpgradeConfig;
+    }
+
+    public static HttpService buildHttpService(Service service) {
+        HttpService httpService = new HttpService(service);
+        Annotation serviceConfigAnnotation = getServiceConfigAnnotation(service);
+
+        if (serviceConfigAnnotation == null) {
+            log.debug("serviceConfig not specified in the Service instance, using default base path");
+            //service name cannot start with / hence concat
+            httpService.setBasePath(HttpConstants.DEFAULT_BASE_PATH.concat(httpService.getName()));
+            return httpService;
+        }
+
+        Struct serviceConfig = serviceConfigAnnotation.getValue();
+
+        httpService.setBasePath(serviceConfig.getStringField(BASE_PATH_FIELD));
+        httpService.setCompressionEnabled(serviceConfig.getBooleanField(COMPRESSION_ENABLED_FIELD));
+        httpService.setCorsHeaders(CorsHeaders.buildCorsHeaders(serviceConfig.getStructField(CORS_FIELD)));
+        httpService.setWebSocketUpgradeConfig(serviceConfig.getStructField(WEBSOCKET_UPGRADE_FIELD));
+
+        List<HttpResource> resources = new ArrayList<>();
+        for (Resource resource : httpService.getBallerinaService().getResources()) {
+            HttpResource httpResource = HttpResource.buildHttpResource(resource, httpService);
+            try {
+                httpService.getUriTemplate().parse(httpResource.getPath(), httpResource,
+                                                   new HttpResourceElementFactory());
+            } catch (URITemplateException | UnsupportedEncodingException e) {
+                throw new BallerinaConnectorException(e.getMessage());
+            }
+            resources.add(httpResource);
+        }
+        httpService.setResources(resources);
+        httpService.setAllAllowedMethods(DispatcherUtil.getAllResourceMethods(httpService));
+
+        return httpService;
+    }
+
+    private static Annotation getServiceConfigAnnotation(Service service) {
+        List<Annotation> annotationList = service.getAnnotationList(HTTP_PACKAGE_PATH,
+                                                                    HttpConstants.ANN_NAME_HTTP_SERVICE_CONFIG);
+
+        if (annotationList == null) {
+            return null;
+        }
+
+        if (annotationList.size() > 1) {
+            throw new BallerinaException(
+                    "multiple service configuration annotations found in service: " + service.getName());
+        }
+
+        return annotationList.isEmpty() ? null : annotationList.get(0);
```

Comment:
Shall we simplify this to 
if (annotationList == null || annotationList.isEmpty()) {
	return null;
}

if (annotationList.size() > 1) {
	throw new BallerinaException(...);
}

return annotationList.get(0);

Reply:
will do